File: src/main/java/com/youdeyiwu/enums/forum/SectionStateEnum.java
package com.youdeyiwu.enums.forum;

import lombok.Getter;

/**
 * section state.
 * If you need to add an enumeration, please append it at the end.
 * This is because the enumeration is stored in natural order, not by name.
 *
 * @author dafengzhen
 */
@Getter
public enum SectionStateEnum {

  /**
   * show (Default).
   */
  SHOW,

  /**
   * hide (Forum Administrator, Access to Section Management).
   */
  HIDE,

  /**
   * lock (The forum administrator, section manager, and users with access keys can access).
   */
  LOCK,

  /**
   * allow (The forum administrator, section administrator, and whitelisted users have access).
   */
  ALLOW,

  /**
   * block (The forum administrators and section moderators have access,
   * while users on the blacklist are not allowed to access).
   */
  BLOCK,

  /**
   * visible after login.
   */
  VISIBLE_AFTER_LOGIN

}

File: app/lib/enums/section_state_enum.dart
import 'package:json_annotation/json_annotation.dart';

enum SectionStateEnum {
  /// show (Default).
  @JsonValue('SHOW')
  show,

  /// hide (Forum Administrator, Access to Section Management).
  @JsonValue('HIDE')
  hide,

  /// lock (The forum administrator, section manager, and users with access keys can access).
  @JsonValue('LOCK')
  lock,

  /// allow (The forum administrator, section administrator, and whitelisted users have access).
  @JsonValue('ALLOW')
  allow,

  /// block (The forum administrators and section moderators have access,
  /// while users on the blacklist are not allowed to access).
  @JsonValue('BLOCK')
  block,

  /// visible after login.
  @JsonValue('VISIBLE_AFTER_LOGIN')
  visibleAfterLogin
}
